0xPolygonHermez/zkevm-bridge-ui
# Fix wrapped token logic

### What does this PR does?

This pull request fixes several bugs and edge cases. It requires an in-depth test session bridging ETH, each of the ERC-20 tokens in the list (they implement different permits) and some custom tokens both native to L1 and L2, back and forth.

Some of the fixes:
1. Incorrectly inferring when a token was native or wrapped and therefore computing incorrect token addresses.
1. Fixing the lack of support for different tokens that share the same address, one in L1 and the other in L2.
1. Wrongly selecting the addresses of L2 native tokens when initiating a bridge. The error happened when checking the `allowance` and doing the approval/permit (@invocamanman bug).
1. Enable permit for DAI
1. Increase claim gas limit to support new tokens deployment on L2
1. Fix MATIC token name

Here is some guidance to test the main relevant aspects:

### How to test 1 and 3?

* Setup the env to work on https://rpc.public.zkevm-test.net (mango)
* Select the zkEVM network
* Add this custom token `0x2A7B29B373CDe8459Dd66e78b56b9c6c1d79Ad96`
* Mint some tokens using the [mint](https://explorer.public.zkevm-test.net/address/0x2A7B29B373CDe8459Dd66e78b56b9c6c1d79Ad96/write-contract#address-tabs) method (6)
* Bridge some tokens to L1
* Bridge some tokens back to L2

### How to test 2?

* Setup the env to work on https://rpc.public.zkevm-test.net (mango)
* Add this token in L1 `0x0B5f1c0E6E3BFEc85e044dB01efc3a4B03d67c69`
* Add this token in L2 `0x0B5f1c0E6E3BFEc85e044dB01efc3a4B03d67c69` (it's the same address but it's a different token)
* Mint some tokens in L2 using the [mint](https://explorer.public.zkevm-test.net/address/0x0B5f1c0E6E3BFEc85e044dB01efc3a4B03d67c69/write-contract#address-tabs) method (6)
* Bridge some tokens to L1
* Bridge some tokens back to L2
* Make sure that both Activity and TX details show the right token.
* Remove the tokens and make sure the right token is removed.
* Make sure that both Activity and TX details show the right info with the tokens removed.

### How to test 4?

* Setup the env to work on https://rpc.public.zkevm-test.net (mango)
* Mint some DAI tokens using the [mint](https://goerli.etherscan.io/address/0xD7E55eB808693D5Ff81a3391c59886C7E0449f35#writeContract) method (4)
* Bridge some DAI to L2
* Bridge some DAI back to L1
